common/network-common/src/main/java/org/apache/spark/network/util/RocksDBProvider.java

```diff
@@ -0,0 +1,170 @@
+ * limitations under the License.
+ */
+
+package org.apache.spark.network.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Objects;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+import org.rocksdb.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.spark.network.shuffledb.StoreVersion;
+
+/**
+ * RocksDB utility class available in the network package.
+ */
+public class RocksDBProvider {
+
+    static {
+      org.rocksdb.RocksDB.loadLibrary();
+    }
+
+    private static final Logger logger = LoggerFactory.getLogger(RocksDBProvider.class);
+
+    public static RocksDB initRockDB(File dbFile, StoreVersion version, ObjectMapper mapper) throws
+        IOException {
+      RocksDB tmpDb = null;
+      if (dbFile != null) {
+        BloomFilter fullFilter =
+          new BloomFilter(10.0D /* BloomFilter.DEFAULT_BITS_PER_KEY */, false);
+        BlockBasedTableConfig tableFormatConfig = new BlockBasedTableConfig()
+          .setFilterPolicy(fullFilter)
+          .setEnableIndexCompression(false)
+          .setIndexBlockRestartInterval(8)
+          .setFormatVersion(5);
+
+        Options dbOptions = new Options();
+        RocksDBLogger rocksDBLogger = new RocksDBLogger(dbOptions);
+
+        dbOptions.setCreateIfMissing(false);
+        dbOptions.setBottommostCompressionType(CompressionType.ZSTD_COMPRESSION);
+        dbOptions.setCompressionType(CompressionType.LZ4_COMPRESSION);
+        dbOptions.setTableFormatConfig(tableFormatConfig);
+        dbOptions.setLogger(rocksDBLogger);
+
+        try {
+          tmpDb = RocksDB.open(dbOptions, dbFile.toString());
+        } catch (RocksDBException e) {
+          if (e.getStatus().getCode() == Status.Code.NotFound) {
+            logger.info("Creating state database at " + dbFile);
+            dbOptions.setCreateIfMissing(true);
+            try {
+              tmpDb = RocksDB.open(dbOptions, dbFile.toString());
+            } catch (RocksDBException dbExc) {
+              throw new IOException("Unable to create state store", dbExc);
+            }
+          } else {
+            // the RocksDB file seems to be corrupt somehow.  Let's just blow it away and create
+            // a new one, so we can keep processing new apps
+            logger.error("error opening rocksdb file {}. Creating new file, will not be able to " +
+              "recover state for existing applications", dbFile, e);
+            if (dbFile.isDirectory()) {
+              for (File f : Objects.requireNonNull(dbFile.listFiles())) {
+                if (!f.delete()) {
+                  logger.warn("error deleting {}", f.getPath());
+                }
+              }
+            }
+            if (!dbFile.delete()) {
+              logger.warn("error deleting {}", dbFile.getPath());
+            }
+            dbOptions.setCreateIfMissing(true);
```

Comment:
This will not take effect if `dbFile.delete()` failed?

Reply:
If `dbFile.delete()` fails, the following `RocksDB.open` will also fail